File: MeleeWeapon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml;

namespace Equipment {

    class MeleeWeapon {

        public string Damage { get; set; }
        public double ArmorDivisor { get; set; }
        public DamageType Type { get; set; }
        public string Reach { get; set; }
        public int Parry { get; set; }
        public bool NoParry { get; set; }
        public bool Fencing { get; set; }
        public bool Unbalanced { get; set; }
        public int Strength { get; set; }
        public bool TwoHanded { get; set; }

        public MeleeWeapon() {
            Damage = "";
            ArmorDivisor = 1.0;
            Type = DamageType.Crushing;
            Reach = "";
            Parry = 0;
            NoParry = false;
            Fencing = false;
            Unbalanced = false;
            Strength = 10;
            TwoHanded = false;
        }

        public MeleeWeapon(XmlNode node) {
            Damage = node.SelectSingleNode("Damage").InnerText;
            ArmorDivisor = double.Parse(node.SelectSingleNode("ArmorDivisor").InnerText);
            Type = DamageType.DamageTypes[node.SelectSingleNode("DamageType").InnerText];
            Reach = node.SelectSingleNode("Reach").InnerText;
            Parry = int.Parse(node.SelectSingleNode("Parry").InnerText);
            NoParry = bool.Parse(node.SelectSingleNode("NoParry").InnerText);
            Fencing = bool.Parse(node.SelectSingleNode("Fencing").InnerText);
            Unbalanced = bool.Parse(node.SelectSingleNode("Unbalanced").InnerText);
            Strength = int.Parse(node.SelectSingleNode("Strength").InnerText);
            TwoHanded = bool.Parse(node.SelectSingleNode("TwoHanded").InnerText);
        }

        public override string ToString() {
            string out = "";

            out += Damage;
            if (ArmorDivisor != 1.0) { out += "(" + ArmorDivisor + ")"; }
            out += DamageType.Code;
            out += ", ";
            out += "Reach " + Reach;
            out += ", ";
            if (!NoParry) {
                out += "Parry " + Parry;
                if (Fencing) { out += "F"; } 
                if (Unbalanced) { out += "U"; }
            } else { out += "No Parry"; }
            out += ", ";
            out += "ST " + Strength;
            if (TwoHanded) { out += "H"; }

            return out;
        }
    }
}

File: com/twilightparadox/triforce700/core/equipment/MeleeWeapon.java
package com.twilightparadox.triforce700.core.equipment;
import com.twilightparadox.triforce700.core.DamageType;

public class MeleeWeapon {

    public String damage;
    public double armorDivisor;
    public DamageType damageType;
    public String reach;
    public int parry;
    public boolean noParry;
    public boolean fencing;
    public boolean unbalanced;
    public int strength;
    public boolean twoHanded;

    public MeleeWeapon() {
        damage = "";
        armorDivisor = 1.0;
        damageType = DamageType.Crushing;
        reach = "";
        parry = 0;
        noParry = false;
        fencing = false;
        unbalanced = false;
        strength = 10;
        twoHanded = false;
    }

    public String toString() {
        String out = "";

        out += damage;
        if (armorDivisor != 1.0) { out += "(" + armorDivisor + ")"; }
        out += damageType.getCode();
        out += ", ";
        out += "Reach " + reach;
        out += ", ";
        if (!noParry) {
            out += "Parry " + parry;
            if (fencing) { out += "F"; } 
            if (unbalanced) { out += "U"; }
        } else { out += "No Parry"; }
        out += ", ";
        out += "ST " + strength;
        if (twoHanded) { out += "H"; }

        return out;
    }
}
